src/client/mod.rs

```diff
@@ -320,26 +336,75 @@ impl<'a, T> VaultClient<'a, T>
     }
 
     /// Revoke `VaultClient`'s token. This token can no longer be used.
+    /// Corresponds to [`/auth/token/revoke-self`][token].
+    ///
+    /// ```no_run
```

Comment:
Instead of making it no_run, what if we generate a new token as a part of this test?